### ISSUE TITLE
Crash on login to server

### ISSUE THREAD

**Issue by BadassOverlord** [Cuppycake/Mech-and-Magic#111]
Hi

Multiple players on my server are having the same crash, including me. Every time a player joins and gets sent to their last location, they are immediately greeted with a crash. It appears to happen on certain co-ordinates where a certain mob spawns. I have to manually reset their position and they can log back in, but if they go back to the same spot, they will crash. Additionally, if someone crashed, they would still remain on the server for a short amount of time. Prior to them actually leaving the server, if any player were to teleport to them, that player would also crash. 

 I asked my players to upload the full log, not just the crash log, so that all information may be present. The crash log is obviously near the end.I attempted to analyze the logs, and have deduced that it has to do with an entity causing a crash. I know it's very generic, but that's all I could manage. Furthermore, I tried to find out which entity it is, by taking multiple crash logs of different scenarios and finding the most common entities. The most common one that is from a mod is "entitytfnagasegment". I'm not entirely sure what's causing the crash.
Here are the crash logs (not the full log because that would be too big):

http://pastebin.com/u874Pg1Z

http://pastebin.com/mbsjJ4TW

http://pastebin.com/veiMv4q1

http://pastebin.com/D55AyWR5

http://pastebin.com/aknBugAy

http://pastebin.com/E2UJ8M1T

http://pastebin.com/RupiGUSh

And here are all the entities from the crash logs compiled into one file:

http://pastebin.com/LL46VHZ9

Thanks

**Comment by Cuppycake** [comment]
Thank you for the in depth report. 
You're using a 1.8 snapshot of bungee cord to run the server?

**Comment by BadassOverlord** [comment]
Hi,
I'm actually running an old custom build of bungeecord that allows me to run both 1.8 servers and 1.7.10 servers. I can't find the source from where I got it from at the moment, but I can upload the jar file and modules.
[Bungee Cord.zip](https://github.com/Cuppycake/Mech-and-Magic/files/344748/Bungee.Cord.zip)

**Comment by Cuppycake** [comment]
Can you replicate the issues using vanilla Forge? This is potentially an issue with molded jar files which we don't build the pack around.

**Comment by BadassOverlord** [comment]
Alright. The next time a user crashes, I'll switch to vanilla forge and ask them to relog.

However, by vanilla forge, do you mean not using KCauldron as well as BungeeCord? Or just removing BungeeCord?

**Comment by Cuppycake** [comment]
Yes a base vanilla unmodded version of Forge.

**Comment by BadassOverlord** [comment]
Still getting the crash, even with both BungeeCord and KCauldron disabled. I'm running the default Forge 1558 and connecting with localhost. I'm still getting the same crash.

Here are the crash logs:
http://pastebin.com/PcHVun0F

Additionally, I found an error on my server log. I've never had anything relating to the crash showing up on the console log other than a player disconnecting because of the crash. 

This is the server log:
http://pastebin.com/yx1iWsus

**Comment by Cuppycake** [comment]
Alright just wanted to make sure it wasn't kcauldron or something before diving into anything.  

This is happening in several different chunks at random?

**Comment by BadassOverlord** [comment]
Yep.
As seen from the logs, the crashes are occurring in several different chunks in several different dimensions. It is entirely random, although there is a pattern of being very far away from the 0,0,0 coordinates. Basically whenever my players go to explore, they may possibly encounter this crash. I've had players trying to look for meteors from AE2 and crashing whilst searching.

Any player that goes to the bugged chunk is also crashed. The bugged chunk is not fixed upon restarting the server, as shown when I switched from KCauldron to vanilla Forge to see if the crash still occurred.

**Comment by Cuppycake** [comment]
Always recommend several backups a day to help combat corrupted chunks.  I'm not home at the moment but I'll dive into it as soon as I can.  Reading the logs is a pain on my phone.

**Comment by BadassOverlord** [comment]
That's why I use KCauldron :)

I can always change AuthMe config to not save players' last location, teleport unauthed to spawn, and force spawn location on join.
I've been using these regularly to reset a player's location when they crash, and it works, but they just can't go back to that area.

And it's fine, take your time, I appreciate your hard work on the modpack, and I can manage until then.

I also tried playing this modpack in June of 2015. We experienced many of the exact same crashes, so unfortunately, we had to switch modpacks.

I thought it was something on my end, but after I actually looked at the logs and Googled far too many things, I decided to report it. :)

Thanks.

**Comment by Cuppycake** [comment]
Yeah it's recently come to my attention that the past 3 versions of the pack have been having issues in certain java environments.  Working on it.  :)  we're also working on 1.9 compilation.

**Comment by Cuppycake** [comment]
True.  But I always prefer an actual backup.  We back up the server every 3 hours with restarts every 4.  Nice little redundant backup to another hdd on my server.  Once the chunk is corrupt it's corrupt.  Even if you move a player it doesn't mean another won't wander into that chunk of  their travels.

**Comment by Cuppycake** [comment]
Have you changed any configs on your server that would effect mob stats? Anything that might change their health or damage values?

**Comment by BadassOverlord** [comment]
I haven't touched any mob stats in the config. The only thing I changed was allowing 9 Enderium essence to form one Enderium ingot from Magical Crops, and allowing the Dig spell from Ars Magica 2. Other than that, I haven't messed with any mob stats from the config.

**Comment by Cuppycake** [comment]
Still can not reproduce this but I believe its a mob error. Any idea what mob might be the the effected area. Surely a modded mob. Maybe ask if anyone saw anything before disconnects? Or if its not a HUGE file Ill take a copy of your world file.

**Comment by BadassOverlord** [comment]
I told my players to tell me what mobs they saw before the crash before I reported it to you. But some didn't remember and most said there wasn't anything weird. One player did, however, tell me about hearing a minecart sound before he crashed.

As for what mob it could be, the crash logs show what entities were in the area before they crashed. I took this section out from each of the crash logs, compiled it together, and ran it through a text analyzer. This shows the most common entities from each of the crashes.

These common entities turned out to be: entityzombie, entitycow, entitycreeper, entitytfnagasegment, 	entityskeleton, 	entityitem, entitytfboar, and entitywitch.

We know it's a mob, so that rules out entityitem. We also know it's a modded mob, so that rules out entityzombie, entitycow, entityskeleton, and entitywitch.

What's left is entitytfnagasegment and entitytfboar. These are the only common modded mobs left. They're both also from the twilight forest mod. But what I don't understand is these mobs don't spawn where my players crash - the overworld. So why are these mobs there? And why are they causing a crash?

Anyway, my world file is 1.38 GB. I'm not sure if that's a huge file or not.

I'm also going to try and disable twilight forest before I go to the corrupted chunk coordinates first.
